1_slices_modules_and_objects/exercises/ordered_vec/src/lib.rs

```diff
@@ -5,17 +5,24 @@ pub struct OrderedVec {
 impl OrderedVec {
     /// Create a new, empty ordered vector
     pub fn new() -> OrderedVec {
-        todo!()
+        Self {
+            vec: Vec::new(),
+        }
     }
 
     /// Add an element to the vector, in order
     pub fn push(&mut self, value: i32) {
-        todo!()
+        self.vec.push(value);
+        let mut i = self.vec.len() - 1;
+        while i > 0 && self.vec[i] < self.vec[i - 1] {
+            self.vec.swap(i, i - 1); 
+            i -= 1;
+        }
```

Comment:
Good! An alternative would be to use `insert` and compute the position at which the new element should reside first, but I believe there would be no advantage in terms of speed or memory. This is a great solution

Reply:
I first did it using a binary search to determine the insert position and then used insert but it would have been at most as good but usually worst since insert already move all element to the right.
We would have save some comparison but I don't that's worse it here